### PR TITLE
implement clean_old_jobs for redis returner

### PR DESCRIPTION
**Pull request by jizhilong**
Implement `clean_old_jobs` for `redis_return` with `redis`'s `EXPIRE` command.
Refs #30513

### CODE REVIEW

File: salt/returners/redis_return.py [hunk 1]
```diff
@@ -189,6 +187,17 @@ def get_minions():
     return list(serv.smembers('minions'))
 
 
+def clean_old_jobs():
```
Comment (rallytime) [inline salt/returners/redis_return.py]: Can you add a docstring to this function?

**Comment by rallytime** [comment]
@jizhilong Thanks for this change. I left one inline comment to fix, and it looks like you also have some pylint errors that need to be addressed before we can get this in: https://jenkins.saltstack.com/job/salt-pr-lint-n/12676/violations/file/salt/returners/redis_return.py/

File: salt/returners/redis_return.py (hunk 1 above)
Comment (jizhilong) [inline salt/returners/redis_return.py]: DONE

**Comment by jizhilong** [comment]
@rallytime thanks for reviewing. I've fixed the pylint errors and added a docstring for `clean_old_jobs`.

**Comment by basepi** [comment]
Go Go Jenkins!

**Comment by jizhilong** [comment]
Hi, I noticed there are some building errors reported by jenkins, but those errors seems not related to the current change, is there something wrong with the building environment?

**Comment by basepi** [comment]
Test failures are unrelated. Thanks for the contribution, @jizhilong!

**Pull request merged**